// guidebox/lib/resources/shows.js
'use strict';

var util = require('util');
var ResourceBase = require('./resourceBase');

var Shows = function(config) {
    ResourceBase.call(this, 'shows', config);
};

util.inherits(Shows, ResourceBase);

(function() {

    this.list = function(params, callback) {
        return this._transmit('GET', null, params, null, callback);
    };

    this.retrieve = function(id, callback) {
        return this._transmit('GET', id, null, null, callback);
    };

    this.images = function(id, params, callback) {
        return this._transmit('GET', id + '/images', params, null, callback);
    };

    this.seasons = function(id, params, callback) {
        return this._transmit('GET', id + '/seasons', params, null, callback);
    };

    this.related = function(id, params, callback) {
        return this._transmit('GET', id + '/related', params, null, callback);
    };

    this.episodes = function(id, params, callback) {
        return this._transmit('GET', id + '/episodes', params, null, callback);
    };

    this.clips = function(id, params, callback) {
        return this._transmit('GET', id + '/clips', params, null, callback);
    };

    this.segments = function(id, params, callback) {
        return this._transmit('GET', id + '/segments', params, null, callback);
    };

    this.availableContent = function(id, params, callback) {
        return this._transmit('GET', id + '/available_content', params, null, callback);
    };


}).call(Shows.prototype);

module.exports = Shows;
