// guidebox/lib/index.js
'use strict';

var resources = require('./resources');
var clientVersion = require('../package.json').version;

var GUIDEBOX_HOST = 'https://api-public.guidebox.com/v2/';
var GUIDEBOX_USERAGENT = 'Guidebox/v2 NodeBindings/' + clientVersion;

var Guidebox = function(apiKey, region = 'US', options) {

    if (!(this instanceof Guidebox)) {
        return new Guidebox(apiKey, options);
    }

    this.resourceBase = require('./resources/resourceBase');

    if (apiKey && typeof apiKey === 'object') {
        options = apiKey;
        apiKey = null;
    }

    this.options = {
        apiKey: apiKey,
        host: GUIDEBOX_HOST,
        userAgent: GUIDEBOX_USERAGENT,
        headers: {
            'user-agent': GUIDEBOX_USERAGENT,
            'Authorization': apiKey,
            'Guidebox-Region': region
        }
    };

    if (options && typeof options === 'object') {

        if (options.hasOwnProperty('apiVersion')) {
            this.options.headers['Guidebox-Version'] = options.apiVersion;
        }

        for (var key in options) {
            this.options[key] = options[key];
        }

    }

    this._initResources();
};

(function() {

    this._initResources = function() {
        var services = Object.keys(resources);

        for (var i = 0; i < services.length; i++) {
            var service = services[i];
            this[service] = new resources[service](this);
        }
    };

}).call(Guidebox.prototype);

module.exports = Guidebox;
